### DISCUSSION TITLE
Content Security Policy directive issues

### DISCUSSION THREAD

**Discussion by robotfx** [FreshRSS/FreshRSS#4348] [Q&A]
I've had to delete my previous installation, that worked just fine, so I can update to the latest version. Now, however, I cannot use inline or external javascripts, such as adsense, for example.
In the browser console there is this:
`Refused to load the script 'https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-xxxxxx' because it violates the following Content Security Policy directive: "script-src 'self' 'nonce-74dd9f7f-e77c-41de-a9e5-fa7754758270'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
`
Any idea how to allow the scripts?

Thank you

**Comment by robotfx** [comment]
nevermind, I've added the following to the layout.phtml and seems to do the job
`header("Content-Security-Policy: default-src 'self' https: data: 'unsafe-inline' 'unsafe-eval' 'unsafe-hashes'");
`

**Reply by Alkarex** [reply]
Such a line basically removes all security provided by CSP so cannot be recommended. Instead, you need to whitelist additional domains you want to allow. See https://developer.mozilla.org/docs/Web/HTTP/CSP

**Comment by Alkarex** [comment]
You can override HTTP headers sent by FreshRSS at your Web server and/or proxy levels, but that will *reduce* security and privacy. This is a better approach than editing FreshRSS code.
For instance:
* Apache: http://httpd.apache.org/docs/trunk/mod/mod_headers.html#header
* Traefik: https://doc.traefik.io/traefik/middlewares/http/headers/#contentsecuritypolicy

See https://developer.mozilla.org/docs/Web/HTTP/CSP
